### PR TITLE
Update compression method and expose badger options

### PR DESCRIPTION
**Pull request by a9kitkumarsinha**
1. Use zlip compression
2. Expose badger options as 
```
badger: 
  levelOneSize: 204800000
  maxLevels: 3
  syncWrites: false
```

2. Add statsd metrics for

- Keys addes
- Keys deleted
- Compaction latency 
- Delete key latency

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -23,6 +23,16 @@ type Config struct {
 	Statsd   *StatsD    `json:"statsd,omitempty" yaml:"statsd" env:"STATSD"`
 	Computed []Computed `json:"computed" yaml:"computed" env:"COMPUTED"`
 	K8s      *K8s       `json:"k8s,omitempty" yaml:"k8s" env:"K8S"`
+	Badger   Badger     `json:"badger,omitempty" yaml:"badger" env:"badger"`
```
Comment (kelindar) [inline internal/config/config.go:26]: `env` should be in caps

File: internal/config/config.go [hunk 2]
```diff
@@ -23,6 +23,16 @@ type Config struct {
...
+}
+
+type Badger struct {
```
Comment (kelindar) [inline internal/config/config.go:29]: Add comments

File: internal/storage/compact/compact.go [hunk 3]
```diff
@@ -134,7 +135,9 @@ func (s *Storage) Compact(ctx context.Context) (interface{}, error) {
 
 	// Wait for the pool to be close
 	close(queue)
-	return wpool.Outcome()
+	out, err := wpool.Outcome()
+	s.monitor.Histogram(ctxTag, "compactlatency", float64(time.Since(st)))
```
Comment (kelindar) [inline internal/storage/compact/compact.go:139]: I thought we had `monitor.Duration` or something

File: internal/config/config.go [hunk 4]
```diff
@@ -23,6 +23,16 @@ type Config struct {
...
+	SyncWrites          *bool   `json:"syncWrites" yaml:"syncWrites" env:"SYNCWRITES"`                            // Whether to sync writes to disk before ack. defaults to true
+	ValueLogMaxEntries  *uint32 `json:"valueLogMaxEntries" yaml:"valueLogMaxEntries" env:"VALUELOGMAXENTRIES"`    // maximum number of entries a value log file can hold approximately. defaults to 5000
+	MaxTableSize        *int64  `json:"maxTableSize" yaml:"maxTableSize" env:"MAXTABLESIZE"`                      // maximum size in bytes for each LSM table or file.
+	LevelOneSize        *int64  `json:"levelOneSize" yaml:"levelOneSize" env:"LEVELONESIZE"`                      // maximum total size in bytes for Level 1. defaults to 1 Million
+	LevelSizeMultiplier *int    `json:"levelSizeMultiplier" yaml:"levelSizeMultiplier" env:"LEVELSIZEMULTIPLIER"` // the ratio between the maximum sizes of contiguous levels in the LSM. defaults to 10
+	MaxLevels           *int    `json:"maxLevels" yaml:"maxLevels" env:"MAXLEVELS"`                               // maximum number of levels of compaction allowed in the LSM. defaults to 7
```
Comment (kelindar) [inline internal/config/config.go:35]: Can we add something like 
```
Defaults string `json:"defaults" yaml:"defaults" env:"DEFAULTS"`
```
which would essentially be one of 3 possible values `badger`, `ingestion`, `storage` and switch either `badger.DefaultOptions(dir)` or 2 other sets of constant values.

I know it's making it a bit specific for our use-cases, but I'd rather not loose the track of the tuned parameters we have and we can interate from here.

File: main.go [hunk 5]
```diff
@@ -91,7 +91,7 @@ func main() {
 		logTable,
 	)
 
-	// onSignal will be called when a OS-level signal is received.
+	// onSignal will be cHandleralled when a OS-level signal is received.
```
Comment (kelindar) [inline main.go:94]: refactor mistake?

File: internal/storage/compact/compact.go (hunk 3 above)
Comment (a9kitkumarsinha) [reply internal/storage/compact/compact.go:139]: We also need the number of times it was called. Duration just stores the time.

**Pull request merged**